packages/@headlessui-react/src/components/transition/transition.tsx

```diff
@@ -463,6 +463,7 @@ function TransitionChildFn<TTag extends ElementType = typeof DEFAULT_TRANSITION_
         // Map data attributes to `entered` class (backwards compatibility)
         !slot.transition && show && entered
       )?.trim() || undefined, // If `className` is an empty string, we can omit it
+    ...transitionDataAttributes(slot),
```

Comment:
I initially wanted to directly return the `data-*` attributes from the `useTransition` hook. However, we want to have `"data-closed": ""` and not `"data-closed": true`. 

This means that in the `Transition` component (where we map the info to the passed in class names) we would have to do something like:

```tsx
    className:
      classNames(
        // Incoming classes if any
        theirProps.className,

        // Apply these classes immediately
        immediate && enter,
        immediate && enterFrom,

        // Map data attributes to `enter`, `enterFrom` and `enterTo` classes
        slot['data-enter'] === '' && enter,
        slot['data-enter'] === '' && slot['data-closed'] === '' && enterFrom,
        slot['data-enter'] === '' && slot['data-closed'] !== '' && enterTo,

        // Map data attributes to `leave`, `leaveFrom` and `leaveTo` classes
        slot['data-leave'] === '' && leave,
        slot['data-leave'] === '' && slot['data-closed'] !== '' && leaveFrom,
        slot['data-leave'] === '' && slot['data-closed'] === '' && leaveTo,

         // Map data attributes to `entered` class (backwards compatibility)
         !slot['data-transition'] === '' && show && entered
       )?.trim() || undefined, // If `className` is an empty string, we can omit it
```

Which felt wrong 😅

Reply:
oh yeah for sure. I think returning the booleans from useTransition is good